haystack/preview/components/file_converters/azure.py

```diff
@@ -0,0 +1,140 @@
+        :param save_json: Whether to save the JSON output of the Azure API. If not set, the value passed to the
+            constructor will be used. Default: `None`
+        :param id_hash_keys: Generate the Document ID from a custom list of strings that refer to the Document's
+            attributes. If you want to ensure you don't have duplicate Documents in your DocumentStore but texts are not
+            unique, you can pass the name of the metadata to use when building the document ID (like
+            `["text", "category"]`) to this field. In this case, the ID will be generated by using the text and the
+            content of the `category` field.
+            If not set, the id_hash_keys passed to the constructor will be used.
+            Default: `None`
+        """
+        id_hash_keys = id_hash_keys or self.id_hash_keys
+        save_json = save_json if save_json is not None else self.save_json
+
+        documents = []
+        for path in paths:
+            path = Path(path)
+            with open(path, "rb") as file:
+                poller = self.document_analysis_client.begin_analyze_document(model_id=self.model_id, document=file)
+                result = poller.result()
+
+            if save_json:
+                with open(path.with_suffix(".json"), "w") as json_file:
+                    json.dump(result.to_dict(), json_file, indent=2)
+
+            file_suffix = path.suffix
+            document = AzureOCRDocumentConverter._convert_azure_result_to_document(result, id_hash_keys, file_suffix)
+            documents.append(document)
+
+        return {"documents": documents}
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serialize this component to a dictionary.
+        """
+        return default_to_dict(
+            self,
+            endpoint=self.endpoint,
+            credential_key=self.credential_key,
+            model_id=self.model_id,
+            save_json=self.save_json,
+            id_hash_keys=self.id_hash_keys,
+        )
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "AzureOCRDocumentConverter":
+        """
+        Deserialize this component from a dictionary.
+        """
+        return default_from_dict(cls, data)
+
+    @staticmethod
+    def _convert_azure_result_to_document(result: AnalyzeResult, id_hash_keys: List[str], file_suffix: str) -> Document:
+        """
+        Convert the result of Azure OCR to a Haystack text Document.
+        """
+        if file_suffix == ".pdf":
+            text = ""
+            for page in result.pages:
+                lines = page.lines if page.lines else []
+                for line in lines:
+                    text += f"{line.content}\n"
+
+                text += "\f"
+        else:
+            text = result.content
+
+        if id_hash_keys:
+            document = Document(text=text, id_hash_keys=id_hash_keys)
+        else:
+            document = Document(text=text)
+
+        return document
```

Comment:
This whole method is quite small and it's used in one place only, so we may even consider moving it back to the run method. Up to you

Reply:
I agree that this method is quite small, but for better readibility, I would still keep it separate. Also, I can see this becoming much more complex once support for table extraction is added (see the corresponding node in v1).